tests/kernel/device/src/abstract_driver.h

```diff
@@ -4,34 +4,45 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef _ABSTRACT_DRIVER_H_
+#define _ABSTRACT_DRIVER_H_
+
 #include <zephyr/ztest.h>
 #include <zephyr/device.h>
+#include <zephyr/sys/check.h>
 
 /* define subsystem common API for drivers */
-typedef int (*subsystem_do_this_t)(const struct device *device, int foo,
-				   int bar);
-typedef void (*subsystem_do_that_t)(const struct device *device,
-				    unsigned int *baz);
-
-struct subsystem_api {
-	subsystem_do_this_t do_this;
-	subsystem_do_that_t do_that;
+typedef int (*abstract_do_this_t)(const struct device *dev, int foo, int bar);
+typedef void (*abstract_do_that_t)(const struct device *dev, unsigned int *baz);
+
+__subsystem struct abstract_driver_api {
```

Comment:
Shouldn't these declaration details (tagging, nomenclature) also be hidden behind a macro?

Reply:
The tags are scanned as text, not after the c preprocessor has expanded macros.

Reply:
Sure, my wrong.